xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-api/src/main/java/org/xwiki/model/reference/DocumentReference.java

```diff
@@ -379,6 +381,33 @@ public DocumentReference withoutLocale()
         return getLocale() != null ? new DocumentReference(this, (Locale) null) : this;
     }
 
+    /**
+     * Retrieve a {@link DocumentReference} from an {@link EntityReference} by extracting it and performing checks.
+     * If the provided entity reference doesn't contain a reference to a document (e.g. it's a wiki reference) then it
+     * returns an empty result.
+     * @param entityReference the reference for which to extract the document reference
+     * @return the extracted {@link DocumentReference} or an empty optional.
+     * @since 16.6.0RC1
+     * @since 16.4.2
+     * @since 15.10.12
+     */
+    @Unstable
+    public static Optional<DocumentReference> valueOf(EntityReference entityReference)
+    {
+        Optional<DocumentReference> result = Optional.empty();
+        if (entityReference != null) {
+            if (entityReference instanceof DocumentReference) {
+                result = Optional.of((DocumentReference) entityReference);
+            } else {
+                EntityReference documentReference = entityReference.extractReference(EntityType.DOCUMENT);
+                if (documentReference != null) {
+                    result = Optional.of(new DocumentReference(documentReference));
```

Comment:
`documentReference` might already be a DocumentReference.

Reply:
I do have a check for that line 399

Reply:
ok